packages/block-library/src/editor.scss

```diff
@@ -109,6 +109,14 @@
 	margin-bottom: $default-block-margin;
 }
 
+.block-editor-block-list__layout > .wp-block:first-child {
+	margin-top: 0;
+}
+
+.block-editor-block-list__layout > .wp-block:last-child {
+	margin-bottom: 0;
+}
```

Comment:
@jasmussen The idea here is that since we're defining the base margin above, we need to reset these margins for the first and last block of each container block.

Reply:
Without this, the padding is not applied properly in the group block (you'll notice an extra space on tthe editor)

Reply:
As can be seen in https://github.com/WordPress/gutenberg/pull/24966#issuecomment-684757681, this isn't working for me. I have the following in my editor style:

```
figure,
p {
	margin-top: 2.5rem;
	margin-bottom: 2.5rem;
}
```

I'm still sort of worried about this being an awfully specific style rule that could have consequences we can't imagine. I don't have a strong opinion, insofar as I appreciate you trying to handle this so I wouldn't have to manually go to the last paragraph and specify a zero margin. I suppose we can try it, and remove it if it has side effects.

Reply:
I'm afraid this had unintended consequences.

![appender](https://user-images.githubusercontent.com/1204802/93855110-3e244300-fcb7-11ea-9c4f-7f2ad4cdb414.gif)

The above happens because every nesting container has a last child that's the appender:

<img width="1094" alt="Screenshot 2020-09-22 at 09 36 51" src="https://user-images.githubusercontent.com/1204802/93855125-47adab00-fcb7-11ea-8fc3-14283fe09b79.png">

Unless, apparently, an adjacent block has been selected:

<img width="1077" alt="Screenshot 2020-09-22 at 09 37 00" src="https://user-images.githubusercontent.com/1204802/93855145-51cfa980-fcb7-11ea-814e-14edd8ebead0.png">

Reply:
I guess that's because the appender div is the "last child" so the margin reset is not applied. I wonder why this empty div is rendered and whether we can remove it.

Reply:
In this PR, I remove the rules: https://github.com/WordPress/gutenberg/pull/25527 — I honestly think it's for the best.

Reply:
I'll take a look but I think there's still value in the rules. So I'd like to see if we can find a better solution.